rag/core/ingest_handler.py

```diff
@@ -56,10 +68,23 @@ def __init__(
         self.database = database
         self.dataset_name = dataset_name
 
-        # Initialize processing logger
-        # Get project directory from config path (parent of llamafarm.yaml)
-        project_dir = self.config_path.parent
-        self.process_logger = ProcessingLogger(str(project_dir), dataset_name)
+        # Extract namespace and project from config path for event logging
+        # Path format: ~/.llamafarm/projects/{namespace}/{project}/llamafarm.yaml
+        # Use LAST occurrence of "projects" to avoid false matches in parent dirs
+        try:
+            path_parts = self.config_path.parts
+            # Find last occurrence of "projects" in path
+            projects_indices = [i for i, part in enumerate(path_parts) if part == "projects"]
+            if projects_indices:
+                projects_idx = projects_indices[-1]  # Use LAST occurrence
+                self.namespace = path_parts[projects_idx + 1] if len(path_parts) > projects_idx + 1 else "default"
+                self.project = path_parts[projects_idx + 2] if len(path_parts) > projects_idx + 2 else "unknown"
+            else:
+                self.namespace = "default"
+                self.project = "unknown"
+        except Exception:
+            self.namespace = "default"
+            self.project = "unknown"
```

Comment:
This is a crazy way to do it. You are leveraging the path structure to determine the project name. You should use the underlying project config file (llamafarm.yaml) to determine namespace and project ID.